src/libs/tryResolveUrlFromApiRoot.ts

```diff
@@ -15,11 +15,8 @@ const ORIGIN_PATTERN = new RegExp(`^(${ORIGINS_TO_REPLACE.join('|')})`);
  * - Similarly for prod or staging URLs we replace the `https://www.expensify`
  * or `https://staging.expensify` part, with `https://{API_ROOT}`
  * - Unmatched URLs (non expensify) are returned with no modifications
- *
- * @param {String} url
- * @returns {String}
  */
-export default function tryResolveUrlFromApiRoot(url) {
+export default function tryResolveUrlFromApiRoot(url: string): string {
     const apiRoot = ApiUtils.getApiRoot({shouldUseSecure: false});
```

Comment:
Don't we need to specify the type here?

Reply:
How is this inference? apiRoot is not set to constant that it's clear to be a string. We are calling ApiUtils.getApiRoot and this does not seem to tell us anything about the type.

Reply:
This is because ApiUtils.getApiRoot is not yet migrated to TS so we can only rely on it's JSDoc for now ([it will be done soon](https://github.com/Expensify/App/issues/24892)). We don't need to be so specific and imo this is enough, Callstack team also accepted this solution. Let me know what you think @s77rt

Reply:
I don't want to block you either. Thanks for your quick response! If the team agrees then it's best to proceed here without further delays.

Reply:
Yeah I thiink it's fine to wait for the `APIUtils` TS migration here